holoviews/element/geom.py

```diff
@@ -50,3 +50,16 @@ class VectorField(Geometry):
 
     vdims = param.List(default=[Dimension('Angle', cyclic=True, range=(0,2*np.pi)),
                                 Dimension('Magnitude')], bounds=(1, None))
+
+
+class Segments(Geometry):
+    """
+    Segments represent a collection of lines in 2D space.
+    """
+    group = param.String(default='Segments', constant=True)
+
+    kdims = param.List(default=[Dimension('x0'), Dimension('y0'),
+                                Dimension('x1'), Dimension('y1')],
+                       bounds=(4, None), constant=True, doc="""
```

Comment:
I think we have decided to keep segments simple, so it should always have 4 dimensions:

```suggestion
                       bounds=(4, 4), constant=True, doc="""
```